[FEATURE] 관리자 계정 생성 및 조회 구현

Romangduck opened the pull request:
*의존성 해결

Romangduck committed:
admin_list

Romangduck committed:
admin_list , detail in Controller -ing

Romangduck committed:
figuring Request

Romangduck committed:
Merge branch 'dev' of https://github.com/donsonioc2010/picasso into feat/a1

Romangduck committed:
feat : temp commit

Romangduck committed:
Merge branch 'dev' into feat/a1

Romangduck committed:
figuring JPA Entity Admin

Romangduck committed:
figuring list.html with thymeleaf

Romangduck committed:
figuring list controller & list service

Romangduck committed:
Merge branch 'dev' of https://github.com/donsonioc2010/picasso into feat/a1

Romangduck committed:
feat : temp commit

Romangduck committed:
git merge dev branch

Romangduck committed:
figuring detail controller , service & html with thymeleaf

Romangduck committed:
figuring Approve

Romangduck committed:
modify for exception and error

Romangduck committed:
finished approve , detail and list figuring

Romangduck committed:
Merge branch 'dev' into feat/a1

Romangduck committed:
Merge branch 'dev' into feat/a1

# Conflicts:
#	Domain/src/main/resources/data.sql

Romangduck committed:
Merge branch 'dev' into feat/a1

Romangduck committed:
dev

Romangduck committed:
feat : 테이블 명 수정에 따른 쿼리 변경

donsonioc2010 committed:
feat : AdminBackEnd완료

Romangduck committed:
Add Modified admin's BackEnd file

approvePicture 메서드에 isSessionUserAdmin 주석처리

Romangduck committed:
Add , modify frontEnd css&html

list.html수정 , adminlist.css ,관리자 계정 생성을 위한 form.html
추가

Romangduck committed:
Add list of Admin and etc for using it

관리자 명단 조회를 위한 메소드 추가

Romangduck committed:
Add Admin Account form & list

관리자 계정 생성 및 리스트

Romangduck committed:
Merge branch 'dev' into feat/a1

# Conflicts:
#	Api/src/main/java/picasso/server/api/admin/controller/AdminController.java
#	Api/src/main/java/picasso/server/api/admin/service/AdminService.java
#	Api/src/main/resources/templates/admin/list.html
#	Domain/src/main/resources/data.sql

Romangduck committed:
Modify html & css, need to revise detail.html&css

html , css 파일 수정 , detail.html&css 개선 필요

Romangduck committed:
Merge branch 'dev' into feat/a1

# Conflicts:
#	Api/src/main/java/picasso/server/api/admin/controller/AdminController.java
#	Domain/src/main/java/picasso/server/domain/domains/user/repository/UserRepository.java
#	Domain/src/main/resources/data.sql

Romangduck committed:
Modify for dependency

의존성 해결을 위한 수정

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+    @GetMapping("/form")
+    public String createForm(HttpSession session, Model model) {
+//        isSessionUserAdmin(session);
+        model.addAttribute("UserDTO", new UserDTO());
```
안필요 삭제

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+
+
+    @GetMapping("/form")
+    public String createForm(HttpSession session, Model model) {
```
Method명 직관적으로 바꾸세요.

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+     * @return
+     */
+    @PostMapping("/form")
+    public String addAdmin(UserDTO dto, HttpSession session) {
+        User user = new User();
+        user.setEmail(dto.getEmail());
+        user.setPassword(dto.getPassword());
+        user.setLoginAt(LocalDateTime.now());
+        user.setNickName(dto.getNickName());
+        user.setUpdatedAt(LocalDateTime.now());
+        user.setProfile(dto.getProfile());
+
+        user.setUserRole(UserRole.ADMIN);
+        userRepository.save(user);
+
+        return "redirect:/";
+    }
```
권한체크를 해야할까요? 안해야 할까요?

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+        user.setUserRole(UserRole.ADMIN);
+        userRepository.save(user);
+
+        return "redirect:/";
```
관리자페이지에서 회원추가를 하는거면 관리자 페이지로 가는게 낫지않을까요?

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+     * @return
+     */
+
+    @GetMapping("/list_admin")
```
`/user/list`  하는건 어떘을까요?

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+        isSessionUserAdmin(session);
+        List<User> adminUsers = adminService.findAllAdmin();
+        model.addAttribute("users", adminUsers);
+        return "admin/list_admin";
```
`admin/memberList` 라는 네이밍은 어때요?

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +61,68 @@ public String list(Model model, HttpSession session) {
+    @GetMapping("/list_admin")
+    public String listAdmin(HttpSession session, Model model) {
+        isSessionUserAdmin(session);
+        List<User> adminUsers = adminService.findAllAdmin();
```
userService에서 UserRole에 따라서 List를 뽑는 기능을 만드는게 좀더 좋은 선택이지 않았을까요?

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/service/AdminService.java:
```diff
@@ -61,4 +65,13 @@ public void approvePicture(Long pictureId) {
     }
+
+
+    @Transactional(readOnly = true)
```
👍

donsonioc2010 commented on Api/src/main/resources/templates/admin/detail.html:
```diff
@@ -1,10 +1,11 @@
 <!DOCTYPE html>
 <html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org">
 <head>
-    <title>Picasso - Picture Detail</title>
+    <title>Picasso - 그림 상세 정보</title>
+    <link rel="stylesheet" href="../static/css/detail.css">
 </head>
 <body>
-<h1>Picture Detail</h1>
+<h1>그림 상세 정보</h1>
 
 <table>
```
우리템플릿 구조있어요~

바꿔

donsonioc2010 commented on Api/src/main/resources/templates/admin/form.html:
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>관리자 계정 생성</title>
+    <link rel="stylesheet" href="../static/css/adminform.css">
+    <style>
+        button {
+            display: block;
+            margin: 20px auto;
+            padding: 10px 20px;
+            font-size: 18px;
+        }
+    </style>
+</head>
+<body>
+<h1>관리자 계정 생성</h1>
+<form th:action="@{/admin/form}" method="post">
+    <label for="email">Email:</label>
+    <input type="email" id="email" name="email" required>
+    <br>
+    <label for="password">Password:</label>
+    <input type="password" id="password" name="password" required>
+    <br>
+    <label for="nickName">Nickname:</label>
+    <input type="text" id="nickName" name="nickName" required>
+    <br>
+    <label for="profile">Profile:</label>
+    <input type="text" id="profile" name="profile" required>
+    <br>
+    <button type="submit">계정 생성</button>
+</form>
+</body>
+</html>
```
템플릿 구조에 맞게 바꿔~

donsonioc2010 commented on Api/src/main/resources/templates/admin/list.html:
```diff
@@ -1,39 +1,45 @@
-            </form>
-        </td>
-    </tr>
-    </tbody>
-</table>
+<div class="container">
+    <h1>관리자 대쉬보드</h1>
+    <table border="1">
+        <thead>
+        <tr>
+            <th>ID</th>
+            <th>그림명</th>
+            <th>시작가</th>
+            <th>생성일</th>
+            <th>승인 상태</th>
+            <th>승인 버튼</th>
+        </tr>
+        </thead>
+        <tbody>
+        <tr th:each="picture : ${pictures}" th:if="${#strings.equals(picture.pictureStatus,'BEFORE_APPROVE')}">
+            <td>
+                <a th:href="@{'/admin/detail/' + ${picture.pictureId}}"><span th:text="${picture.pictureId}"></span></a>
+            </td>
+            <td>
+                <a th:href="@{'/admin/detail/' + ${picture.pictureId}}"><span th:text="${picture.pictureName}"></span></a>
+            </td>
+            <td th:text="${picture.startingPrice}"></td>
+            <td th:text="${picture.bidStartDate}"></td>
+            <td th:text="${picture.pictureStatus}"></td>
+            <td>
+                <form th:action="@{'/admin/approve/' + ${picture.pictureId}}" method="post">
+                    <button type="submit">Approve</button>
+                </form>
+            </td>
+        </tr>
+        </tbody>
+    </table>
+</div>
 </body>
 </html>
```
타임리프 템플릿 구조에 맞게 `바꿔`

donsonioc2010 commented on Api/src/main/resources/templates/admin/list_admin.html:
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <div class="image-container">
+        <img src="/static/asset/logo.png" alt="logo" height="100px"/>
+        <title>Picasso - 승인 요구 리스트</title>
+        <link rel="stylesheet" href="../static/css/adminlist.css">
+    </div>
+</head>
+<body>
+<h1>관리자 목록</h1>
+<table border="1">
+    <thead>
+    <tr>
+        <th>ID 번호</th>
+        <th>이메일</th>
+        <th>최근 로그인 시간</th>
+        <th>최근 수정 시간</th>
+        <th>별명</th>
+        <th>프로필 링크</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr th:each="user : ${users}">
+        <td th:text="${user.id}"></td>
+        <td th:text="${user.email}"></td>
+        <td th:text="${user.loginAt}"></td>
+        <td th:text="${user.updatedAt}"></td>
+        <td th:text="${user.nickName}"></td>
+        <td th:text="${user.profile}"></td>
+    </tr>
+    </tbody>
+</table>
+</body>
+</html>
```
타임리프 템플릿 구조에 맞게 `바꿔`

donsonioc2010 reviewed (commented):
html 바꿔.

Romangduck committed:
Modify code reviewing

사용할 Dto 만 적용가능 하게끔 SignUpAdminRequestDto 작성후 적용
Build패턴-> 해당 Dto만 적용
높은 가시성을 위한 메서드 네이밍 및 경로 재설정

Romangduck committed:
안ìì안쓰는 ì의존설성 정리

donsonioc2010 commented on Api/src/main/java/picasso/server/api/admin/controller/AdminController.java:
```diff
@@ -53,19 +64,69 @@ public String list(Model model, HttpSession session) {
+    @GetMapping("/user/list")
+    public String AdminList(HttpSession session, Model model) {
+        isSessionUserAdmin(session);
+        List<User> adminUsers = adminService.findAllAdmin();
+        model.addAttribute("users", adminUsers);
```
```suggestion
        model.addAttribute("users", adminService.findAllAdmin());
```

donsonioc2010 reviewed (approved):
HTML은 나보고 하라?
